inngest/src/main/kotlin/com/inngest/State.kt

```diff
@@ -20,6 +29,20 @@ class State(
         return sb.toString()
     }
 
+    private fun findNextAvailableStepId(id: String): String {
+        var stepNumber = 1
+        var possibleStepId: String
+        while (true) {
+            possibleStepId = "$id:$stepNumber"
+            if (possibleStepId !in stepIds) {
+                break
+            }
+            stepNumber++
+        }
```

Comment:
not sure if there's a better way to find the next available `stepNumber` but looks like this [hasn't been a problem in the inngest-js SDK](https://github.com/inngest/inngest-js/blob/79069e1a3d700624ce49b323922c113fc952bcc6/packages/inngest/src/components/execution/v1.ts#L823-L829) yet so leaving it as is for now

Reply:
the Go SDK does something like this.
https://github.com/inngest/inngestgo/blob/0a00daba0b2db68ff0f080f787cf63f0a63b44d8/internal/sdkrequest/manager.go#L123-L131

Reply:
doing a loop every time to find the next possible one might not be the most efficient. probably having a map or something of the kind to track state temporarily as it goes through the existing state would be ideal

Reply:
cool, that's the algorithm I was envisioning anyway. updated to store count in a map